src/st/no3/46684.c

```diff
@@ -1268,8 +1268,93 @@ INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntitySplashWater);
 // splash animation and sound when Alucard touches water
 INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntitySurfacingWater);
 
-// small water droplets go to the side
-INCLUDE_ASM("asm/us/st/no3/nonmatchings/46684", EntitySideWaterSplash);
+// ID 0x37
+void EntitySideWaterSplash(Entity* self) {
+    Primitive* prim;
+    s16 primIndex;
+    s32 temp_lo;
+    s32 temp_s0;
+    s32 temp_s1;
+    s32 temp_s3;
+    u16 params;
+    s16 angle;
+    u16 x, y;
+    s32* ptr;
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(D_80180B00);
+        primIndex = g_api.AllocPrimitives(PRIM_GT4, 1);
+        if (primIndex != -1) {
+            prim = &g_PrimBuf[primIndex];
+            self->primIndex = primIndex;
+            self->flags |= FLAG_HAS_PRIMS;
+            while (prim != NULL) {
+                prim->u0 = prim->u2 = 0xF0;
+                prim->u1 = prim->u3 = 0xFF;
+                prim->v0 = prim->v1 = 0;
+                prim->v2 = prim->v3 = 0xF;
+                prim->clut = 0x161;
+                prim->tpage = 0x1A;
+                prim->r2 = prim->g2 = prim->b2 = prim->r3 = prim->g3 =
+                    prim->b3 = prim->r0 = prim->g0 = prim->b0 = prim->r1 =
+                        prim->g1 = prim->b1 = 128;
+                prim->p1 = 0;
+                prim->priority = self->zPriority + 2;
+                prim->blendMode = 0x37;
+                prim = prim->next;
+            }
+            params = self->params;
+            temp_s0 = params & 0xF;
+            if (temp_s0 == 0) {
+                g_api.PlaySfx(D_801813A8[0]);
+            }
+            angle = LOH(D_801838E4[(params >> 3) & 0x1E]);
```

Comment:
Have you tried D_801838E4 as a struct?

Reply:
Why would i? this is an AND operation, not a multiplication, & 0x1E does not suggest a struct of any kind. If it's something that i'm not seeing please make a scratch with the suggested theory.